src/devices/bus/spectrum/neogs.cpp

```diff
@@ -153,8 +190,90 @@ template <u8 Bank> void neogs_device::ram_bank_w(offs_t offset, u8 data)
 	m_ram->write((0xc000 * Bank) + offset, data);
 }
 
+void neogs_device::spi_ctrl_w(u8 data)
+{
+	if (BIT(data, 7))
+		m_spi_ctrl |= (data & 0x0f);
+	else
+		m_spi_ctrl &= (~data & 0x0f);
+}
+
+u8 neogs_device::sd_r()
+{
+	u8 din = m_spi_data_in_latch;
+	if (!machine().side_effects_disabled())
+		sd_w(0xff);
+
+	return din;
+}
+
+void neogs_device::sd_w(u8 data)
+{
+	m_spi_data_out = data;
+#if TIMINGS_PERFECT
+	m_spi_clock_cycles = 8;
+	m_spi_clock->adjust(m_maincpu->clocks_to_attotime(1) / 4, 0, m_maincpu->clocks_to_attotime(1) / 4);
+#else
+	m_sdcard->spi_ss_w(BIT(~m_spi_ctrl, 0));
+	for (u8 m = 0x80; m; m >>= 1)
+	{
+		m_sdcard->spi_mosi_w(m_spi_data_out & m ? 1 : 0);
+		m_sdcard->spi_clock_w(CLEAR_LINE);
+		m_sdcard->spi_clock_w(ASSERT_LINE);
+	}
+#endif
+}
+
+void neogs_device::dac_flush()
+{
+	s16 left = 0;
+	s16 right = 0;
+	for(auto ch = 0; ch < 8; ch++)
+	{
```

Comment:
This `for` loop isn’t formatted like the others (the others have a space, like `for (...)`).